src/popup/pages/NetworkForm.vue

```diff
@@ -46,112 +46,140 @@
   </div>
 </template>
 
-<script>
-import { mapActions, mapGetters } from 'vuex';
-import { ROUTE_NETWORK_SETTINGS } from '../router/routeNames';
+<script lang="ts">
+import {
+  computed, defineComponent, onMounted, ref,
+} from '@vue/composition-api';
+import { ROUTE_NETWORK_EDIT, ROUTE_NETWORK_SETTINGS } from '../router/routeNames';
 import { defaultNetwork } from '../utils';
+import { useDispatch, useGetter } from '../../composables/vuex';
+import type { INetworkBase } from '../../types';
+
 import BtnMain from '../components/buttons/BtnMain.vue';
 import InputField from '../components/InputField.vue';
 import PlusCircleIcon from '../../icons/plus-circle.svg?vue-component';
 
+interface INetworkForm extends INetworkBase {
+  index?: number
+}
+
 export const NETWORK_PROPS = {
-  url: null,
-  name: null,
+  url: '',
+  name: '',
   middlewareUrl: defaultNetwork.middlewareUrl,
   compilerUrl: defaultNetwork.compilerUrl,
   backendUrl: defaultNetwork.backendUrl,
 };
 
-export default {
+export default defineComponent({
   components: {
     BtnMain,
     InputField,
   },
-  data() {
-    return {
-      PlusCircleIcon,
-      newNetwork: {
-        ...NETWORK_PROPS,
+  setup(props, { root }) {
+    const networks = useGetter('networks');
+    const selectNetwork = useDispatch('selectNetwork');
+
+    const newNetwork = ref<INetworkForm>({
+      ...NETWORK_PROPS,
+    });
+
+    const formConfig = [
```

Comment:
I suggest to add types here, mostly to bind the `key` with the `INetworkBase` properties.